akka-http/src/main/scala/akka/http/javadsl/server/directives/PathDirectives.scala

```diff
@@ -136,52 +66,213 @@ abstract class PathDirectives extends MiscDirectives {
+  }
+  def pathPrefix[T1, T2](p: PathMatcher2[T1, T2], inner: BiFunction[T1, T2, Route]): Route = ScalaRoute {
+    D.pathPrefix(p.toScala) { (t1, t2) ⇒ inner.apply(t1, t2).toScala }
+  }
+
+  /**
+   * Applies the given matcher directly to a prefix of the unmatched path of the
+   * [[RequestContext]] (i.e. without implicitly consuming a leading slash).
+   * The matcher has to match a prefix of the remaining path.
+   * If matched the value extracted by the PathMatcher is extracted on the directive level.
+   */
+  def rawPathPrefix(segment: String, inner: Supplier[Route]): Route = ScalaRoute {
+    D.rawPathPrefix(segment) { inner.get().toScala }
+  }
+  def rawPathPrefix(pm: PathMatcher0, inner: Supplier[Route]): Route = ScalaRoute {
+    D.rawPathPrefix(pm.toScala) { inner.get().toScala }
+  }
+  def rawPathPrefix[T1](pm: PathMatcher1[T1], inner: Function[T1, Route]): Route = ScalaRoute {
+    D.rawPathPrefix(pm.toScala) { t1 ⇒ inner.apply(t1).toScala }
+  }
+  def rawPathPrefix[T1, T2](pm: PathMatcher2[T1, T2], inner: BiFunction[T1, T2, Route]): Route = ScalaRoute {
+    D.rawPathPrefix(pm.toScala) { case (t1, t2) ⇒ inner.apply(t1, t2).toScala }
+  }
+
+  /**
+   * Checks whether the unmatchedPath of the [[RequestContext]] has a prefix matched by the
+   * given PathMatcher. In analogy to the `pathPrefix` directive a leading slash is implied.
+   */
+  def pathPrefixTest(segment: String, inner: Supplier[Route]): Route = ScalaRoute {
+    D.pathPrefixTest(segment) { inner.get().toScala }
+  }
+  def pathPrefixTest(pm: PathMatcher0, inner: Supplier[Route]): Route = ScalaRoute {
+    D.pathPrefixTest(pm.toScala) { inner.get().toScala }
+  }
+  def pathPrefixTest[T1](pm: PathMatcher1[T1], inner: Function[T1, Route]): Route = ScalaRoute {
+    D.pathPrefixTest(pm.toScala) { t1 ⇒ inner.apply(t1).toScala }
+  }
+  def pathPrefixTest[T1, T2](pm: PathMatcher2[T1, T2], inner: BiFunction[T1, T2, Route]): Route = ScalaRoute {
+    D.pathPrefixTest(pm.toScala) { case (t1, t2) ⇒ inner.apply(t1, t2).toScala }
+  }
+
+  /**
+   * Checks whether the unmatchedPath of the [[RequestContext]] has a prefix matched by the
+   * given PathMatcher. However, as opposed to the `pathPrefix` directive the matched path is not
+   * actually "consumed".
+   */
+  def rawPathPrefixTest(segment: String, inner: Supplier[Route]): Route = ScalaRoute {
+    D.rawPathPrefixTest(segment) { inner.get().toScala }
+  }
+  def rawPathPrefixTest(pm: PathMatcher0, inner: Supplier[Route]): Route = ScalaRoute {
+    D.rawPathPrefixTest(pm.toScala) { inner.get().toScala }
+  }
+  def rawPathPrefixTest[T1](pm: PathMatcher1[T1], inner: Function[T1, Route]): Route = ScalaRoute {
+    D.rawPathPrefixTest(pm.toScala) { t1 ⇒ inner.apply(t1).toScala }
+  }
+  def rawPathPrefixTest[T1, T2](pm: PathMatcher2[T1, T2], inner: BiFunction[T1, T2, Route]): Route = ScalaRoute {
+    D.rawPathPrefixTest(pm.toScala) { case (t1, t2) ⇒ inner.apply(t1, t2).toScala }
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Non ascii characters are not allowed](https://www.codacy.com/app/vasseurjb/akka/file/2046270054/issues/source?bid=3132306&fileBranchId=3132379#l174)